libs/s25main/ai/random.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2005 - 2025 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "helpers/random.h"
+#include <random>
+
+namespace AI {
+
+std::minstd_rand& getRandomGenerator();
+
+template<typename T>
+T randomValue(T min = std::numeric_limits<T>::min(), T max = std::numeric_limits<T>::max())
+{
+    return helpers::randomValue(getRandomGenerator(), min, max);
+}
+
+// Return a random bool:
+//   random()           ... will return true|false with 50% chance each
+//   random(15)         ... will return true in 1/15 of the cases
+inline bool random(unsigned total = 1u)
+{
+    return helpers::randomValue(getRandomGenerator(), 0u, total) == 0u;
+}
+
+//   random(20, 5)      ... will return true in ~3/4 of the cases (random(20) > 5)
+inline bool random(unsigned total, unsigned chance)
+{
+    return helpers::randomValue(getRandomGenerator(), 0u, total) > chance;
```

Comment:
- For 20, 5 we get values 0, 1, 2, ..., 20, i.e. 21 values 
- We succeed for all but 6 (0..5)
- --> Chance of 15 out of 21

So basically the semantic isn't "succeed for chance out of total cases" but "fail for chance out of total cases" (with an off-by-one error). IMO this is hard to understand at the call site.

```suggestion
    return helpers::randomValue(getRandomGenerator(), 1u, total) <= chance;
```

- Values 1, 2, ... 20, i.e. 20 values
- Succeed for 1,2,3,4,5
- --> 5 out of 20